packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -945,7 +945,7 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
 
   @override
   Iterable<OverlayEntry> createOverlayEntries() sync* {
-    yield new OverlayEntry(builder: _buildModalBarrier);
+    yield new OverlayEntry(builder: _buildModalBarrier, semanticsBarrier: true);
```

Comment:
looking at this it's sad that we can't have the modal barrier itself somehow occlude all the semantics below it rather than having it done in the overlay. but i don't see how we could sanely do that.